### PR TITLE
fix(cd): Workflow and Dependency Updates

### PR DESCRIPTION
**Pull request by zFernand0**
## Proposed changes

- Update workflows to run on LTS versions of Node.js
- Add OVSX CLI as a dev dependency
- Update overrides object to pnpm's format

## Release Notes

Milestone: `vNext`

Changelog: Workflow and dependency updates

## Types of changes

What types of changes does your code introduce to Zowe Explorer?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Updates to Documentation or Tests (if none of the other choices apply)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This checklist will be used as reference for both the contributor and the reviewer_

- [x] I have read the [CONTRIBUTOR GUIDANCE](https://github.com/zowe/vscode-extension-for-zowe/wiki/Best-Practices:-Contributor-Guidance) wiki
- [x] PR title follows [Conventional Commits Guidelines](https://www.conventionalcommits.org/en/v1.0.0-beta.2/)
- [x] PR Description is included
- [ ] gif or screenshot is included if visual changes are made
- [ ] `yarn workspace vscode-extension-for-zowe vscode:prepublish` has been executed
- [ ] All checks have passed (DCO, Jenkins and Code Coverage)
- [ ] I have added unit test and it is passing
- [ ] I have added integration test and it is passing
- [ ] There is coverage for the code that I have added
- [ ] I have tested it manually and there are no regressions found
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any PR dependencies have been merged and published (if appropriate)

## Further comments

We never had the OVSX cli installed as a devDependency, it was just because of `npx` that this worked. In order to avoid undesired dependency version updates (with `pnpm dlx`), we use `pnpm exec` which won't download the latest version of the package from npm 😋

### CODE REVIEW

File: packages/zowe-explorer/package.json [hunk 1]
```diff
@@ -1958,7 +1958,7 @@
     "createDemoNodes": "ts-node ./scripts/createDemoNodes.ts"
   },
   "engines": {
-    "vscode": "^1.63.0"
+    "vscode": "^1.73.0"
```
Comment (JillieBeanSim) [inline packages/zowe-explorer/package.json:1961]: was just glancing over and notice this, changing to 1.79.0 for readonly capability in this [PR](https://github.com/zowe/vscode-extension-for-zowe/pull/2505/files)

**Comment by traeok** [comment]
I've been investigating the failing Theia tests and have found the source of the issue.

Commits 272ef96fa9ffaf28942fa8d626734ecb4c97423f and before work fine in Theia - localized strings show as expected.

When testing commit 3edd22fd8676fc60504f163b3f4de4e04b46432d, the strings in Theia are broken. Any commit after this one exhibits the same behavior. This seems related to the `package.json` import statement in `ZoweLocalStorage.ts`.

Since this involves some refactoring of the local storage work, it would be best to address this in a separate PR.

**Review by JillieBeanSim — APPROVED** [review]
LGTM! thanks @zFernand0 for this important fix and updates

**Review by traeok — APPROVED** [review]
LGTM! And thanks for updating the core VS Code dependencies @zFernand0 😋

**Pull request merged**